hivemind/optim/averaged.py

```diff
@@ -105,6 +168,55 @@ def _average_parameters_in_background(
             except Exception as e:
                 logger.error(f"Averaging round failed: caught {e}.")
 
+    @property
+    def is_synchronized(self) -> bool:
+        return self.local_epoch + self.max_allowed_epoch_difference >= self.decentralized_state.max_epoch
```

Comment:
Why not make this discrepancy in terms of just steps rather than epochs? We won't need to introduce one more entity this way

Reply:
The problem with steps is that one cannot properly aggregate them in a swarm of peers where anyone can crash/leave at any time.

Epochs were introduced specifically to address this issue by introducing some shared "global steps": if all peers collectively run 10000 steps, we increment epoch counter by one globally; if any of the peers leave after the epoch is incremented, their local steps will disappear from the DHT, but it will not affect the global epochs.